running-code/execute-script.md

```diff
@@ -0,0 +1,191 @@
+Linux, macOS, all BSDs, and many other operating systems are fully- or mostly-POSIX compliant.
+
+Windows is not natively POSIX compliant. However, some "modes" inside of Windows are, such as [WSL](https://learn.microsoft.com/en-us/windows/wsl/about)
+(Windows Subsystem for Linux), gitbash, or some VSCode terminals.
+:::
+
+## Windows executables
+
+If your Windows machine has Python registered as the default application associated with `.py` files, then any Python
+scripts can be run as commands. However, only one Python can be registered at a time, so all Python scripts run this
+way will use the same Python environment.
+
+Additionally, most Windows Python installs come with the [Python Launcher](https://docs.python.org/3/using/windows.html#python-launcher-for-windows)
+which, in addition to allowing specifying the version of Python to run, can also read shebang lines and emulate some of that behavior.
+This allows for shebang lines to be re-used between Linux, macOS, and Windows systems. However, on Windows the command must still
+be prefaced with another command (`py`).
+
+```bash
+py my_program.py
+# ✨ Hello from Python ✨
+```
+
+:::{tip}
+While all Python files should end in a `.py`, this naming is necessary for Windows to associate a script with Python, as opposed
+to Linux where `.py` is a convention and the shebang associates the file with Python.
+
+Also, although there is no in-source format that can tell Windows what to do with a Python file, executing a
+Python file with a shebang on Windows also does not cause any issues. Python just sees the whole line as
+a comment and ignores it!
+
+Because of these differences it is best practice to use both a shebang and `.py` for all Python scripts.
+:::
+
+## Executable comparisons
+
+### Pros of passing a file to `python`:
+- don't need execute permissions
+- works on every system
+- explicit about what you expect to happen
+
+### Pros of inserting a shebang to the file:
+- file is associated with specific python
+  - don't have to remember which
+- don't have to use the `python` command
+  - don't have to even remember it is a Python script
+
+
+## Separating script from import behavior
+
+Sometimes a Python file that is useful to execute is also useful to import. You may want to use `shiny_hello`
+in another Python file. But right now, the `my_program.py` does all its script behavior even when it is imported. Consider
+
+```python
+import my_program
+
+def guess_my_number():
+    my_program.shiny_hello()
+    print("Was your number 42?")
+
+guess_my_number()
+# ✨ Hello from Python ✨
+# ✨ Hello from Python ✨
+# Was your number 42?
+```
+
+You may not have expected it to print the hello twice, but it did. This is because `my_program` is set to
+_always_ call `shiny_hello`, and now `guess_my_number` also calls it. That's two times. How can we make
+`my_program` only call `shiny_hello` when it is used as a script?
```

Comment:
```suggestion
You may not have expected it to print the hello twice, but it did. The line prints twice because `my_program` is designed to
_always_ call `shiny_hello`. Now `guess_my_number` also calls `shiny_hello`. 

```python
def shiny_hello():
    print("\N{Sparkles} Hello from Python \N{Sparkles}")
    
# my_program runs the shiny_hello function
shiny_hello()
```

This is why the line prints twice!  How can you make
`my_program` only call `shiny_hello` when it is run as a script?
```

Reply:
I think your suggestion was mis-typed?